src/cli/global/install.rs

```diff
@@ -145,11 +145,43 @@ async fn setup_environment(
             let env_dir = EnvDir::from_env_root(project.env_root.clone(), env_name.clone()).await?;
             let prefix = Prefix::new(env_dir.path());
             let prefix_package = prefix.find_designated_package(package_name).await?;
-            for (executable_name, _) in prefix.find_executables(&[prefix_package]) {
-                let mapping =
-                    Mapping::new(ExposedName::from_str(&executable_name)?, executable_name);
+            let package_executables = prefix.find_executables(&[prefix_package]);
+            for (executable_name, _) in &package_executables {
+                let mapping = Mapping::new(
+                    ExposedName::from_str(executable_name)?,
+                    executable_name.clone(),
+                );
                 project.manifest.add_exposed_mapping(env_name, &mapping)?;
             }
+            // If no executables were found, automatically expose the package name itself from the other packages.
+            // This is useful for packages like `ansible` and `jupyter` which don't ship executables their own executables.
+            if !package_executables
+                .iter()
+                .any(|(name, _)| name.as_str() == package_name.as_normalized())
+            {
+                let installed_packages = prefix.find_installed_packages(None).await?;
+                for package in &installed_packages {
+                    let executables = prefix.find_executables(&[package.clone()]);
+
+                    // Check if any of the executables match the package name
+                    if let Some(executable) = executables
+                        .iter()
+                        .find(|(name, _)| name.as_str() == package_name.as_normalized())
+                    {
+                        let mapping = Mapping::new(
+                            ExposedName::from_str(&executable.0)?,
+                            executable.0.clone(),
+                        );
+                        project.manifest.add_exposed_mapping(env_name, &mapping)?;
+                        tracing::warn!(
+                            "Automatically exposing `{}` from package `{}`",
+                            package_name.as_normalized(),
+                            package.repodata_record.package_record.name.as_normalized()
+                        );
+                        break; // Stop once we found the match
+                    }
+                }
```

Comment:
Maybe extract that into its own function

Reply:
Did that :+1: